src/engraving/libmscore/cmd.cpp

```diff
@@ -728,6 +731,23 @@ void Score::addInterval(int val, const std::vector<Note*>& nl)
 
         select(note, SelectType::SINGLE, 0);
     }
+    unsigned i = 0;
+    for (Note* on : nl) {
+        if (on->tieFor()) {
+            auto it = find(nl.begin(), nl.end(), on->tieFor()->endNote());
+            if (it != nl.end()) {
+                unsigned index = it - nl.begin();
```

Comment:
Note that it is not necessary to use this `index` variable here. If you have an iterator, you can just dereference that iterator:
```c++
    Note* oldEndNote = *it;
```
But in this case, that is not even necessary either, because you already have a pointer to that note: it's just `on->tieFor()->endNote()`.

Reply:
Thank you for the feedback, I made the other changes you suggested.
For this one, I'm using the indices of the tied notes in the original list to find the corresponding notes in `newNotes` so I can tie them together, which is why I am using find rather than contains.

I considered getting rid of the vector entirely and using `on->chord()->upNote()` and `on->tieFor()->endNote()->chord()->upNote()` instead, of course changed to `downNote` if the interval is negative, but this does not work if notes in the middle of a chord are selected individually. Please correct me if I'm misunderstanding your feedback.

Reply:
Ah, you're right. You need `index` to retrieve the correct note from the other list. Looks good now!